quesma/queryparser/pancake_aggregation_parser_metrics.go

```
@@ -67,6 +67,18 @@ func generateMetricSelectedColumns(ctx context.Context, metricsAggr metricsAggre
 			innerFieldsAsSelect = append(innerFieldsAsSelect, model.NewColumnRef(metricsAggr.SortBy))
 		}
 		return innerFieldsAsSelect, nil
+	case "rate":
+		if len(metricsAggr.Fields) > 0 {
+			switch metrics_aggregations.NewRateMode(ctx, metricsAggr.mode) {
+			case metrics_aggregations.RateModeSum:
+				result = []model.Expr{model.NewFunction("sumOrNull", getFirstExpression())}
+			case metrics_aggregations.RateModeValueCount:
+				result = []model.Expr{model.NewCountFunc(getFirstExpression())}
+			default:
+				// should never happen because of parsing checks
+				logger.ErrorWithCtx(ctx).Msgf("unknown rate mode: %s", metricsAggr.mode)
```

We should return the error here.